assets/src/edit-story/components/canvas/canvasLayout.js

```diff
@@ -1,62 +1,223 @@
 /**
  * External dependencies
  */
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 /**
  * Internal dependencies
  */
-import { CENTRAL_RIGHT_PADDING, LEFT_NAV_WIDTH, PAGE_WIDTH, PAGE_HEIGHT, PAGE_NAV_PADDING } from '../../constants';
-import Page from './page';
+import { LEFT_NAV_WIDTH, PAGE_WIDTH, PAGE_HEIGHT, PAGE_NAV_BUTTON_WIDTH } from '../../constants';
+import { useStory } from '../../app';
+import useCanvas from './useCanvas';
+import PageDisplay from './page';
+import PageFrames from './pageFrames';
+import PageEdit from './pageEdit';
 import PageMenu from './pagemenu';
 import PageNav from './pagenav';
 import Carrousel from './carrousel';
 import SelectionCanvas from './selectionCanvas';
 
-const Background = styled.div`
+// QQQQ: remove.
+const DEBUG_ANGLE = 0;
+
+const WorkspaceBackground = styled.div`
 	background-color: ${ ( { theme } ) => theme.colors.bg.v1 };
-	display: flex;
-	flex-direction: column;
-	align-items: center;
-	justify-content: center;
+	width: 100%;
 	height: 100%;
+	position: relative;
+	user-select: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: darkgray;
+		backface-visibility: hidden;
+		transform-style: preserve-3d;
+		transform-origin: 50% 50%;
+		transform: perspective(2500px) scale(0.4) rotateY(${ DEBUG_ANGLE }deg);
+	` ) }
+`;
+
+const Layer = css`
+	position: absolute;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+`;
 
+// @todo: the menu and carousel heights are not correct until we make a var-size
+// page.
+const Layout = css`
 	display: grid;
 	grid:
-    ".    .      .          . .        ." 1fr
-    ".    prev   page       . next     ." ${ PAGE_HEIGHT }px
-    ".    .      menu       . .        ." 48px
-    ".    .      carrousel  . .        ." 1fr
-    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_PADDING }px 1fr ${ CENTRAL_RIGHT_PADDING }px;
+    ".    .      . .          . .        ." 1fr
+    ".    prev   . page       . next     ." ${ PAGE_HEIGHT }px
+    ".    .      . menu       . .        ." 48px
+    ".    .      . .          . .        ." 1fr
+    ".    carrousel      carrousel carrousel  carrousel carrousel        ." 60px
```

Comment:
feel free to ignore this as this isn't really part of this PR, but my OCD demands that we fix `carousel` misspelling everywhere ;)

Reply:
Also covered by #4106 FWIW

Reply:
Will sync it.